### ISSUE TITLE
Update CCTZ

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#42252]
See https://github.com/google/cctz/pull/233

**Comment by ddelnano** [comment]
@alexey-milovidov I'm interested in picking up some beginner issues.

I've been trying to get clickhouse to build as a first step before anything else. I've followed the documentation that was mentioned [here](https://github.com/ClickHouse/ClickHouse/issues/42110#issuecomment-1273113887), but it seems many of the third party components generate executables that fail on the target machine clickhouse is being built for (as mentioned in my [later comment](https://github.com/ClickHouse/ClickHouse/issues/42110#issuecomment-1274123082)).

I worked around this by building `protoc` from the clickhouse/protobuf repo manually, but this appears to happen for many other dependencies (`llvm-tblgen` and `grpc_cpp_plugin` also experience this issue).

The overall process I followed was the following:
- Create vagrant VM running Ubuntu jammy
- Install all the dependencies as specified [here](https://clickhouse.com/docs/en/development/build) and opted to use clang 15
- Cloned the clickhouse repository pulling in the submodules and attempt the build
```
$ git clone --recursive https://github.com/ClickHouse/ClickHouse.git
$ cd ClickHouse
$ mkdir build
$ cd build
$ cmake ..
$ ninja
```

Are you able to confirm that those instructions should work out of the box? I would appreciate if you could help me understand what I'm doing wrong to build the repo.

**Comment by ddelnano** [comment]
I've opened a PR for this (linked above). I need some assistance ensuring that I test it properly (due to my build issues), so any assistance with that would be greatly appreciated.

**Comment by alexey-milovidov** [comment]
After https://github.com/ClickHouse/cctz/pull/5 is merged, 
now the remaining step is a pull request to update the submodule reference.

**Comment by ddelnano** [comment]
@alexey-milovidov I understand that clickhouse is flexible to be built in many environments, but as a first time contributor I was trying to create a consistent environment once I ran into issues on my first build.

> The problem you encountered in Vagrant is most likely caused by Vagrant itself.
"Exec format error": either the disk size is too low, or the wrong architecture was accidentally specified.

The vagrant VM has plenty of disk space, so I'm fairly confident it's an architecture problem. Looking at the verbose ninja output for the first target to use `protoc`, it appears that the correct architecture is passed through (`--target`, `--sysroot` and `--gcc-toolchain` are using x86_64). However, you can see that `file` doesn't recognize the binary format when compilation finishes.

```
vagrant@ubuntu-jammy:/vagrant/build$ ninja -v contrib/arrow-cmake/orc_proto.pb.h
[0/2] /usr/bin/cmake -P /vagrant/build/CMakeFiles/VerifyGlobs.cmake
[1/2] : && /usr/bin/clang++-15 --target=x86_64-linux-gnu --sysroot=/vagrant/cmake/linux/../../contrib/sysroot/linux-x86_64/x86_64-linux-gnu/libc --gcc-toolchain=/vagrant/cmake/linux/../../contrib/sysroot/linux-x86_64 -std=c++20 -fdiagnostics-color=always -Xclang -fuse-ctor-homing -fsized-deallocation  -gdwarf-aranges -mssse3 -msse4.1 -msse4.2 -mpclmul -mpopcnt -fasynchronous-unwind-tables -ffile-prefix-map=/vagrant=. -falign-functions=32 -mbranches-within-32B-boundaries -fdiagnostics-absolute-paths -fstrict-vtable-pointers -fexperimental-new-pass-manager -w -O2 -g -DNDEBUG -O3 -g -gdwarf-4  -fno-pie --gcc-toolchain=/vagrant/cmake/linux/../../contrib/sysroot/linux-x86_64 --ld-path=/vagrant/build/ld.lld -rdynamic -Wl,--gdb-index -Wl,--build-id=sha1 -no-pie -Wl,-no-pie    -Xlinker --no-undefined contrib/protobuf-cmake/CMakeFiles/protoc.dir/__/protobuf/src/google/protobuf/compiler/main.cc.o -o contrib/protobuf-cmake/protoc  contrib/protobuf-cmake/lib_libprotoc.a  contrib/protobuf-cmake/lib_libprotobuf.a  -lpthread  contrib/zlib-ng-cmake/lib_zlib.a  -Wl,--start-group  base/glibc-compatibility/libglibc-compatibility.a  base/glibc-compatibility/memcpy/libmemcpy.a  contrib/libcxx-cmake/libcxx.a  contrib/libcxxabi-cmake/libcxxabi.a  contrib/libunwind-cmake/libunwind.a  -Wl,--end-group  -nodefaultlibs /usr/lib/llvm-15/lib/clang/15.0.3/lib/linux/libclang_rt.builtins-x86_64.a  -lc -lm -lrt -lpthread -ldl && :
[2/2] cd /vagrant/build/contrib/arrow-cmake && /vagrant/build/contrib/protobuf-cmake/protoc -I /vagrant/contrib/orc/c++/../proto --cpp_out="/vagrant/build/contrib/arrow-cmake" /vagrant/contrib/orc/c++/../proto/orc_proto.proto
FAILED: contrib/arrow-cmake/orc_proto.pb.h contrib/arrow-cmake/orc_proto.pb.cc /vagrant/build/contrib/arrow-cmake/orc_proto.pb.h /vagrant/build/contrib/arrow-cmake/orc_proto.pb.cc
cd /vagrant/build/contrib/arrow-cmake && /vagrant/build/contrib/protobuf-cmake/protoc -I /vagrant/contrib/orc/c++/../proto --cpp_out="/vagrant/build/contrib/arrow-cmake" /vagrant/contrib/orc/c++/../proto/orc_proto.proto
/bin/sh: 1: /vagrant/build/contrib/protobuf-cmake/protoc: Exec format error
ninja: build stopped: subcommand failed.

# Verify executable format is not recognized
vagrant@ubuntu-jammy:/vagrant/build$ file /vagrant/build/contrib/protobuf-cmake/protoc
/vagrant/build/contrib/protobuf-cmake/protoc: data

# See that the VM has plenty of disk space
vagrant@ubuntu-jammy:/vagrant/build$ df -h
Filesystem      Size  Used Avail Use% Mounted on
tmpfs           393M  956K  392M   1% /run
/dev/sda1        49G  9.1G   40G  19% /
tmpfs           2.0G     0  2.0G   0% /dev/shm
tmpfs           5.0M     0  5.0M   0% /run/lock
vagrant         932G  723G  210G  78% /vagrant
tmpfs           393M  4.0K  393M   1% /run/user/1000
```

Thanks for merging https://github.com/ClickHouse/cctz/pull/5 and I have created https://github.com/ClickHouse/ClickHouse/pull/42273 for the final change.

**Comment by alexey-milovidov** [comment]
For some reason, it was updated to 2022d instead of 2022e.
I will do the final update.

**Comment by ddelnano** [comment]
It appears I opened my PR 2 hours before upstream merged the 2022e support. So when I merged upstream's master branch it missed the 2022e update.

<img width="967" alt="Screen Shot 2022-10-16 at 9 27 08 PM" src="https://user-images.githubusercontent.com/5855593/196070816-6a311b8c-b94b-4969-933c-da610912ea07.png">
<img width="801" alt="Screen Shot 2022-10-16 at 9 27 18 PM" src="https://user-images.githubusercontent.com/5855593/196070819-52a99249-0605-43d8-b9f8-de800bf7d0ad.png">



Sorry about that and thanks for updating it to the latest release.
